### PR TITLE
ComponentJS: make JogWheelBasic correctly switch which deck it controls

### LINKED ISSUE #11867: JogWheelBasic component doesn't take into account deck switching
### Bug Description

As seen [here](https://github.com/mixxxdj/mixxx/blob/dfde28dbe44375706abcf8846e7fd099c1c66a87/res/controllers/midi-components-0.0.js#L772) the `JogWheelBasic` component doesn't take into account the selected deck/group when scratching.

I could try a PR if you want. I assume the fix would be to add special handling [here](https://github.com/mixxxdj/mixxx/blob/dfde28dbe44375706abcf8846e7fd099c1c66a87/res/controllers/midi-components-0.0.js#L707) similar to how it's done for `EffectAssignmentButton`.

### Version

2.3.5

### OS

Arch Linux btw

### PR DESCRIPTION
**Pull request by miwig**
by getting the deck from the `group` in `connect`. Fixes #11867. Please let me know if any changes are needed @Swiftb0y.

### CODE REVIEW

File: res/controllers/midi-components-0.0.js [hunk 1]
```diff
@@ -789,6 +789,10 @@
         input: function(_channel, control, _value, status, _group) {
             throw "Called wrong input handler for " + status + ": " + control + ".\n" +
                 "Please bind jogwheel-related messages to inputWheel and inputTouch!\n";
+        },
+        connect: function() {
+            Component.prototype.connect.call(this);
+            this.deck = parseInt(script.channelRegEx.exec(this.group)[1]);
         }
```
Comment (Swiftb0y) [inline res/controllers/midi-components-0.0.js:796]: I think this needs an explanation.
```suggestion
        // this is needed for features such as "deck switching" that work
        // by changing the component group. It is assumed they call `connect`
        // afterwards.
        connect: function() {
            Component.prototype.connect.call(this);
            this.deck = parseInt(script.channelRegEx.exec(this.group)[1]);
        }
```

**Review by Swiftb0y — COMMENTED** [review]
Small request. Also, we need you to sign the [Mixxx Contributor Agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ) and comment here when you have done so. It gives us permission to distribute your contribution under the GPL v2 or later license and the Apple Mac App Store. It is also helpful for us to have contact information for contributors in case we may need it in the future. Thank you.

**Comment by miwig** [comment]
> Small request. Also, we need you to sign the [Mixxx Contributor Agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ) and comment here when you have done so. It gives us permission to distribute your contribution under the GPL v2 or later license and the Apple Mac App Store. It is also helpful for us to have contact information for contributors in case we may need it in the future. Thank you.

Signed.

**Review by Swiftb0y — APPROVED** [review]
Thank you.

**Pull request merged**